src/Util/Annotation/DocBlock.php

```diff
@@ -101,7 +101,10 @@ public static function ofClass(\ReflectionClass $class): self
         );
     }
 
-    public static function ofMethod(\ReflectionMethod $method): self
+    /**
+     * @psalm-param class-string $className
+     */
+    public static function ofMethod(\ReflectionMethod $method, string $className = null): self
```

Comment:
Now there are two ways to create DocBlock object
```
DocBlock::ofClass(
    new \ReflectionClass($class)
);
DocBlock::ofMethod(
    new \ReflectionMethod($class, $method)
);
```
Unfortunately, can not obtain class namespace from method reflection object, only class that declares method.
Maybe it's better to describe only one method to create a DocBlock object e.g.
```
public static function fromReflection(\ReflectionClass $class, \ReflectionMethod $method = null): self
```
or make a public constructor
```
public function __construct(\ReflectionClass $class, \ReflectionMethod $method = null)
```
@Ocramius what do you think about that?

Reply:
What about separate named ctors (as separate static methods)?

Also, please keep the ctor private: that's what gave us this flexibility in first place 👍

Reply:
There is a difference between docblocks of classes and those of methods right? I think the current structure makes that quite clear?

Reply:
Yeah, all good with that